### ISSUE TITLE
Simple Date rule is not working. Says "TypeError: parseISO is not a function"

### ISSUE THREAD

**Issue by multiojuice** [mikeerickson/validatorjs#369]
![image](https://user-images.githubusercontent.com/23535069/83664470-0d00f500-a598-11ea-9d73-bbabf79f4eab.png)

rule is simple {date: 'required|date'} and data is {date: '05-12-1999'}

version {    "validatorjs": "^3.18.1" }

I have also tried other formatting like '05/12/1999', but it seems no good. Am I misusing? thanks

**Comment by mikeerickson** [comment]
Can you please try installing `date-fns` package directly.  This dependency should have been installed along with validatorJS package.

**Comment by drthoben** [comment]
I was able to reproduce the error. Installing the `date-fns` package manually did not solve the problem for me.

My versions:
```
validatorjs: "^3.17.1", installed: 3.18.1
date-fns: "^2.14.0", installed: 2.14.0
```

**Comment by mikeerickson** [comment]
@drthoben hmm, ok I will have a look at this further.  This has been a reported issue in the past, but the install of date-fns manually cured the problem.  I won’t be able to look at this until the weekend.

**Comment by drthoben** [comment]
No worries @mikeerickson. Is there anything I can help you with solving the issue?

**Comment by mikeerickson** [comment]
@drthoben i let you know as soon as it is ready so you can test against your implementation.

**Comment by darkovasic** [comment]
Same problem here. Manually installing date-fns didn't solve the issue.

"validatorjs": "^3.18.1"
"date-fns": "^2.14.0"

date format: 2020-07-23

Recently I've updated validators to the latest version because of some issues with translations. I think that date validation worked before the update, but can't be 100% sure about it.

```
react-dom.production.min.js:21 Uncaught TypeError: parseISO is not a function
    at isValidDate (rules.js:22)
    at Rule.after_or_equal [as fn] (rules.js:379)
    at Rule._apply (rules.js:487)
    at Rule.validate (rules.js:469)
    at eval (validator.js:114)
    at Validator.checkAsync (validator.js:137)
    at Validator.fails (validator.js:506)
    at eval (AddEvent.js:153)
    at Object.da (react-dom.production.min.js:14)
    at ka (react-dom.production.min.js:15)
```

**Comment by mikeerickson** [comment]
This issue will be resolved in the upcoming 3.19 release

**Comment by mikeerickson** [comment]
@multiojuice I have not been able to reproduce this issue locally.  Would you mind create a small test application and either email to me (codedungeon@gmail.com) or post it on git or somewhere where I can download and try.

**Comment by mikeerickson** [comment]
@multiojuice Hey there, just pinging you again to see if you can provide the above information.  I am ready to push out a new release, but I want to make sure this issue is resolved as I am not able to reproduce it locally.

**Comment by Nodios** [comment]
This just happened to me. I am using react/mobx/mobx-react-form. I have form with input type date, and rules for that field: 'required|date'. I have set the default value(using moment) and it works as expected, but when I change the input value, then it breaks with the above error. 

Here are my deps:
```
    "mobx": "5.15.4",
    "mobx-react": "6.2.2",
    "mobx-react-form": "2.0.8",
    "moment": "2.27.0",
    "oidc-client": "1.10.1",
    "react": "16.13.1",
    "react-app-polyfill": "1.0.6",
    "react-dev-utils": "10.2.1",
    "react-dom": "16.13.1",
    "react-router-dom": "5.2.0",
    "validatorjs": "3.18.1"
```

**Comment by mikeerickson** [comment]
@nodios This is a pretty difficult edge case to debug. Since it works locally on new installations (using node) 

Any chance you can create a demo which shows the issue?

**Comment by Nodios** [comment]
@mikeerickson I will try tonight

**Comment by Nodios** [comment]
@mikeerickson here [repro](https://github.com/Nodios/validatorjs-369-issue-repro)

**Comment by mikeerickson** [comment]
@Nodios great, thank you so much I will have a look as soon as possible, likely on Saturday

**Comment by GaborGyorgy** [comment]
@mikeerickson Any update on this? I have crated [this](https://codesandbox.io/s/nameless-frost-snocp?file=/src/index.ts) sandbox with `date-fns` (latest) manually installed and I had no luck.

**Comment by mikeerickson** [comment]
@GaborGyorgy Two items

1. I dont have TypeScript setup locally for immediate testing, thus I will have to task that for another day
2. Using on a Vue test application, if I use

```js
import * as Validator from 'validatorjs' 
```

All works as expected.  Can you attempt to format as above in your TS application

I will get babel-node setup locally so I can test ES6 import from node, but as stated it works in a VueJS application which is using `import` format as outlined above.

**Comment by mikeerickson** [comment]
@GaborGyorgy 

OK, I have just created a quick and dirty app using node / babel and have tried your code and all works fine without any modification

After reviewing your codebox, I see you are using some typing file `@ v3.15`. I am not sure if this is the culprit, but I can say, the offending code was new as of `3.17` so it *may* bet related to the typings file.

I am trying to extract the referenced typings file from the sandbox, but not having much success. If you don't mind, please send me a copy of what you have so I can analyze it locally.  

Finally, if you attempt to use the library as a commonJS format, does it have any positive impact?

```js
const Validator = require('validatorjs')
```

I am not sure if the `Rules` access will work within TypeScript, if for any other reason than I am not familiar with TypeScript

```js
import {Rules} from 'validatorjs'

vs

const {Rules} = require('validatorjs')
```

**Comment by kennethsumang** [comment]
I was able to replicate the issue above.
```
"date-fns": "^2.16.1",
"validatorjs": "^3.18.1"
```

I was using `import * as Validator from 'validatorjs';` to import the package and the data to validate contains a simple date string `"2020-09-23"`.

![image](https://user-images.githubusercontent.com/17355092/93951359-aa1ab000-fd78-11ea-9bb3-6dc5e0ec3d1c.png)

**Comment by mikeerickson** [comment]
@nethzsumang can you please test with latest version `3.19.2` as this is the version I am testing against. 

Also, are you using with TypeScript?

**Comment by kennethsumang** [comment]
@mikeerickson I updated the package to the latest version `3.19.2` and the issue persists.

No, I'm not using TypeScript.

**Comment by mikeerickson** [comment]
@nethzsumang I am trying in vein to reproduce this issue.  I am sure it exists as it has been reported numerous times, but it always works here.

Would you be able to create a simple test app which exhibits the issue?

**Comment by kennethsumang** [comment]
@mikeerickson I have created [this sandbox](https://codesandbox.io/s/crimson-bash-qo0gu?file=/src/index.js) for the reproduction of the issue.

**Comment by mikeerickson** [comment]
@nethzsumang alas I see the issue. The date-fns module is not imported correctly when using in browser. 

All this time I have been testing against a node application. 

I will get a fix created and post a new version as soon as possible.

**Comment by mikeerickson** [comment]
@nethzsumang @GaborGyorgy @Nodios @drthoben @multiojuice 

Issue **finally** resolved in 3.20 release! 🎊

https://github.com/skaterdav85/validatorjs/releases/tag/v3.20.0
